lib/modules/manager/puppet/puppetfile-parser.ts

```diff
@@ -0,0 +1,106 @@
+import { newlineRegex, regEx } from '../../../util/regex';
+import type { PuppetForgeUrl, PuppetfileModule } from './types';
+
+const forgeRegex = regEx(/^forge\s+['"]([^'"]+)['"]/);
+const commentRegex = regEx(/#.*$/);
+
+/**
+ * For us a Puppetfile is build up of forges that have Modules.
+ *
+ * Modules are the updatable parts.
+ *
+ */
+export class Puppetfile {
+  private forgeModules: Map<PuppetForgeUrl, PuppetfileModule[]> = new Map<
+    PuppetForgeUrl,
+    PuppetfileModule[]
+  >();
+
+  public add(currentForge: PuppetForgeUrl, module: PuppetfileModule): void {
+    if (Object.keys(module).length === 0) {
+      return;
+    }
+
+    if (!this.forgeModules.has(currentForge)) {
+      this.forgeModules.set(currentForge, []);
+    }
+
+    this.forgeModules.get(currentForge)?.push(module);
+  }
+
+  public getForges(): PuppetForgeUrl[] {
+    return Array.from(this.forgeModules.keys());
+  }
+
+  public getModulesOfForge(forgeUrl: any): PuppetfileModule[] {
+    const modules = this.forgeModules.get(forgeUrl);
+
+    return modules || [];
```

Comment:
please use ?? operator 
```suggestion
    return modules ?? [];
```